### PR TITLE
First task

### PR DESCRIPTION
**Pull request by velychkodmytro**

### CODE REVIEW

File: dist/adapter.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+"use strict";
```
Comment (deogonit) [inline dist/adapter.js]: Please remove directory dist/*
You don't need save compiled js code from ts code

File: index.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
```
Comment (deogonit) [inline index.html]: why do you need this file???

File: routes/dfaf/user.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"name":"Dima","age":20}
```
Comment (deogonit) [inline routes/dfaf/user.json]: Please remove this file. It's not secret, but it's not normal safe user data

File: src/adapter.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import { v4 as uuidv4 } from 'uuid'
```
Comment (deogonit) [inline src/adapter.ts]: `import { v4 as uuid }` more better

File: src/adapter.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  createFolderIfNotExist,
+  createContent,
+  getContent,
+  deleteContent,
+} from './workingWithData'
+
+type UserInfoType = { name: string; age: number }
+
+class WorkingWithData {
+  pathToFile: string
```
Comment (deogonit) [inline src/adapter.ts]: `private readonly pathFile: string`
`private readonly nameFile: string` more better

File: src/app.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+
```
Comment (deogonit) [inline src/app.ts]: ??????

File: src/adapter.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    await deleteContent(path, nameOfFile)
+  }
+}
+
+const user = new WorkingWithData('./routes/dfaf', 'user.json')
+
+user.init({ name: 'Dima', age: 20 })
+user.create({ name: 'Alex', age: 30 })
+//user.get()
+//user.delete('./routes/dfaf', 'user.json')
```
Comment (deogonit) [inline src/adapter.ts]: remove unnecessary comments

File: src/adapter.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+  // private readonly BodyaLox:string
+
+  constructor(pathToFile: string, fileName: string) {
+    this.pathToFile = pathToFile
+    this.fileName = fileName
+    // this.content = content
+  }
+
+  async init(data: UserInfoType) {
```
Comment (deogonit) [inline src/adapter.ts]: you need also specify what your function will return. in this case you don't return anything, so you write this
`async init(data: UserInfoType): Promise<void> {`

Comment (deogonit) [reply src/adapter.ts]: Please add return value for all functions below
https://stackoverflow.com/questions/59664670/typescript-async-function-return-type-void-vs-promisevoid****

File: src/adapter.ts [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+  async get() {
+    await getContent(this.pathToFile, this.fileName)
+  }
+
+  async delete(path: string, nameOfFile: string) {
+    await deleteContent(path, nameOfFile)
+  }
+}
+
+const user = new WorkingWithData('./routes/dfaf', 'user.json')
```
Comment (deogonit) [inline src/adapter.ts]: class called WorkingWithData, but variable has been named user. it's not clear

File: src/models/user.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import * as fs from 'fs'
+import { v4 as uuidv4 } from 'uuid'
+
+//НЕ ИСПОЛЬЗУЮ НИГДЕ
```
Comment (deogonit) [inline src/models/user.ts]: remove

File: src/models/user.ts [hunk 11]
```diff
@@ -0,0 +1,19 @@
+import * as fs from 'fs'
```
Comment (deogonit) [inline src/models/user.ts]: it's not correct to import all functions from library
its info from Python, but I guess it works also in NodeJS 
https://stackoverflow.com/questions/2386714/why-is-import-bad :)
https://stackoverflow.com/questions/36795819/when-should-i-use-curly-braces-for-es6-import/36796281

File: src/workingWithData.ts [hunk 12]
```diff
@@ -0,0 +1,54 @@
+import { v4 as uuidv4 } from 'uuid'
+import { resolve } from 'path'
+import { readFile, open, writeFile, unlink, access, mkdir } from 'fs/promises'
+
+type UserInfoType = { name: string; age: number }
+
+export const createFolderIfNotExist = async function (path: string) {
+  const pathToDIr: string = resolve(path)
```
Comment (deogonit) [inline src/workingWithData.ts:8]: For me 4 indents more better, than 2

File: src/workingWithData.ts [hunk 13]
```diff
@@ -0,0 +1,54 @@
+import { v4 as uuidv4 } from 'uuid'
+import { resolve } from 'path'
+import { readFile, open, writeFile, unlink, access, mkdir } from 'fs/promises'
+
+type UserInfoType = { name: string; age: number }
```
Comment (deogonit) [inline src/workingWithData.ts]: You can move this type to another directory. For example you can create directory types and file user.type.ts and put there your type. Also you can put there validators

File: src/workingWithData.ts [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+) {
+  const takePath: string = resolve(path, fileName)
+
+  try {
+    await access(takePath)
+    await writeFile(takePath, JSON.stringify(content))
+    console.log('Content has been added')
+  } catch (error) {
+    await open(takePath, 'a+')
+    console.log('File was created')
```
Comment (deogonit) [inline src/workingWithData.ts:30]: your final code should not contains a console.log. but you can use custom logger

File: src/workingWithData.ts [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+  const pathToDIr: string = resolve(path)
+  try {
+    await access(pathToDIr)
+  } catch (error) {
+    await mkdir(pathToDIr)
+    console.log('The dir has been created')
+  }
+}
+
+export const createContent = async function (
```
Comment (deogonit) [inline src/workingWithData.ts]: Please specify return type

Comment (deogonit) [reply src/workingWithData.ts]: also you in this case more better don't use arrow function
`export async function createContent(...`

File: src/workingWithData.ts [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+    await open(takePath, 'a+')
+    console.log('File was created')
+  }
+}
+
+export const getContent = async function (path: string, fileName: string) {
+  const filePath: string = resolve(path, `${fileName}`)
+
+  try {
+    const fileContent: string = await readFile(filePath, 'utf-8')
```
Comment (deogonit) [inline src/workingWithData.ts:41]: You don't use this variable in you code below. Please return value, if reading file was succesfull
https://nodejs.org/api/fs.html#fs_filehandle_readfile_options

File: .gitignore [hunk 17]
```diff
@@ -0,0 +1,2 @@
+node_modules
+.vscode
```
Comment (deogonit) [inline .gitignore]: add please **dist** directory

File: src/adapter.ts [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+import {
+  createFolderIfNotExist,
+  createContent,
+  getContent,
+  deleteContent,
+} from './workingWithData'
+
+type UserInfoType = { name: string; age: number }
+
+class WorkingWithData {
```
Comment (deogonit) [inline src/adapter.ts:11]: Mb `class User`

File: src/models/items.ts [hunk 19]
```diff
@@ -0,0 +1,17 @@
+import * as fs from 'fs'
+import { v4 as uuidv4 } from 'uuid'
+//import user from '../models/user'
+
+//НЕ ИСПОЛЬЗУЮ НИГДЕ
```
Comment (deogonit) [inline src/models/items.ts]: ??????

**Review by deogonit — CHANGES_REQUESTED** [review]
Need changes. Remove also comments from configs. You should specify return type and more practice with ts
https://www.typescriptlang.org/docs/handbook/typescript-from-scratch.html

File: dist/adapter.js (hunk 1 above)
Comment (velychkodmytro) [reply dist/adapter.js]: Don't understand you

File: index.html (hunk 2 above)
Comment (velychkodmytro) [reply index.html]: done

File: routes/dfaf/user.json (hunk 3 above)
Comment (velychkodmytro) [reply routes/dfaf/user.json]: done

File: src/adapter.ts (hunk 4 above)
Comment (velychkodmytro) [reply src/adapter.ts]: Done

File: src/adapter.ts (hunk 5 above)
Comment (velychkodmytro) [reply src/adapter.ts]: Done

File: src/app.ts (hunk 6 above)
Comment (velychkodmytro) [reply src/app.ts]: Done

File: src/adapter.ts (hunk 7 above)
Comment (velychkodmytro) [reply src/adapter.ts]: Done

File: src/adapter.ts (hunk 8 above)
Comment (velychkodmytro) [reply src/adapter.ts]: Done

File: src/adapter.ts (hunk 9 above)
Comment (velychkodmytro) [reply src/adapter.ts]: I will use this class for Items also, so i can't name it class User. Using this class for Entities. Can name it Entities

File: src/models/user.ts (hunk 10 above)
Comment (velychkodmytro) [reply src/models/user.ts]: Removed "//НЕ ИСПОЛЬЗУЮ НИГДЕ"

File: src/models/user.ts (hunk 11 above)
Comment (velychkodmytro) [reply src/models/user.ts]: Done

File: src/workingWithData.ts (hunk 12 above)
Comment (velychkodmytro) [reply src/workingWithData.ts:8]: ))))))))) Done

File: src/workingWithData.ts (hunk 13 above)
Comment (velychkodmytro) [reply src/workingWithData.ts]: Done

File: src/workingWithData.ts (hunk 14 above)
Comment (velychkodmytro) [reply src/workingWithData.ts:30]: What a custom logger?

File: src/workingWithData.ts (hunk 15 above)
Comment (velychkodmytro) [reply src/workingWithData.ts]: also you in this case more better don't use arrow function ----- Also would be better if you don't use arrow function in this case
Done

File: src/workingWithData.ts (hunk 16 above)
Comment (velychkodmytro) [reply src/workingWithData.ts:41]: Not done))))) 
image
image
Vs code bagged and creating dfaf dir every time when i start. It is using my last code which doesn't exists now........

File: .gitignore (hunk 17 above)
Comment (velychkodmytro) [reply .gitignore]: Done

File: src/adapter.ts (hunk 18 above)
Comment (velychkodmytro) [reply src/adapter.ts:11]: What about class Entity?

File: src/models/items.ts (hunk 19 above)
Comment (velychkodmytro) [reply src/models/items.ts]: Removed. Done

File: src/workingWithData.ts (hunk 14 above)
Comment (deogonit) [reply src/workingWithData.ts:30]: winston or pino for example

File: src/workingWithData.ts (hunk 16 above)
Comment (deogonit) [reply src/workingWithData.ts:41]: Remove vs Code and use normal IDE or find place where you create dir dfaf

File: dist/adapter.js (hunk 1 above)
Comment (deogonit) [reply dist/adapter.js]: It's not correct to push compiled and builded code. Your builded code have been saved in directory **dist**.

File: src/adapter.ts (hunk 9 above)
Comment (deogonit) [reply src/adapter.ts]: In this class you work with file sytem, so name FileAdapter would be better

File: routes/dfaf/user.json (hunk 3 above)
Comment (deogonit) [reply routes/dfaf/user.json]: you have not removed this file

File: src/workingWithData.ts (hunk 12 above)
Comment (deogonit) [reply src/workingWithData.ts:8]: I see 2 indents

**Review by deogonit — CHANGES_REQUESTED** [review]
See unresolved comments

**Review by deogonit — APPROVED** [review]
lgtm

**Pull request merged**